### PR TITLE
Feature/ms update

### PR DESCRIPTION
**Pull request by nick-j-roberts**
- Adds plugin which looks up s3 docs created during SST modeling runs, adds id, category, and tropical storms attributes and modifies metadata to include link to png url for SST output, then outputs final doc in JSON format
  - output can be consumed by meilisearch to add or update documents backing the stormviewer application
- Revises meilisearch upload method

### CODE REVIEW

**Comment by nick-j-roberts** [comment]
Things to consider:
- the methodology I use requires a JSON document containing tropical storm data to be on s3
  - I think this makes sense because @ahdietrich sent me JSON data specific to Trinity watershed and transposition region, so this should not be made a constant or anything like that
- I do not know how dynamic the front-end design of the stormviewer application is, so it is unclear if updating the meilisearch database to have an additional filterable attribute of tropical storms is going to require changes on the front end to utilize this filterable attribute

**Comment by nick-j-roberts** [comment]
I just changed the tropical storm data for the meilisearch document as having a list of objects associated with each document would make it difficult to use in filters, which is what I would think the purpose of this data is. It is easy to revert to a list of objects, but would need clarification on how this would be interacted with on meilisearch

File: stormcloud/plugins/ms_update/ms_update_plugin.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import boto3
+from create_ms_docs import SSTGeom, SSTMeta, SSTS3Document, SSTStart, SSTStats, create_ms_documents, sanitize_for_s3
+
+PLUGIN_PARAMS = {
+    "required": [
+        "watershed_name",
+        "transposition_domain",
+        "duration",
+        "s3_bucket",
+        "tropical_storm_json_s3_uri",
```
Comment (slawler) [inline stormcloud/plugins/ms_update/ms_update_plugin.py]: Can we make this an optional parameter?

**Review by slawler — CHANGES_REQUESTED** [review]
Please add a readme with the following:

1. A small example of the contents of the `tropical_storm_json_s3_uri` file for reference.
2. A couple example records of the output file.

Lets also write the output to s3, and include the path as optional (with a default setting of `s3://{bucket}/watersheds/{waterhsed_name}/{waterhsed_name}-transpo-area-v01/{waterhsed_name}-ranked-events.json`).

File: stormcloud/plugins/ms_update/ms_update_plugin.py (hunk 1 above)
Comment (nick-j-roberts) [reply stormcloud/plugins/ms_update/ms_update_plugin.py]: These requests should be taken care of now

**Comment by nick-j-roberts** [comment]
I have renamed the plugin from ms_update to doc_rank to reflect that the utility of the plugin is not solely geared towards meilisearch consumption

**Comment by nick-j-roberts** [comment]
As a side note, the docs from the Trinity SST runs are missing the watershed name. This causes issues with path name creation in the documents, but I can add an override to pull the watershed name from provided plugin parameters instead of the document to take care of this.

Example of missing data from s3 docs for Trinity (s3://tempest/watersheds/trinity/trinity-transpo-area-v01/72h/docs/19790201.json):
```JSON
"metadata": {
    ...
    "watershed_name": "",
    ...
  }
```

**Comment by nick-j-roberts** [comment]
> As a side note, the docs from the Trinity SST runs are missing the watershed name. This causes issues with path name creation in the documents, but I can add an override to pull the watershed name from provided plugin parameters instead of the document to take care of this.
> 
> Example of missing data from s3 docs for Trinity (s3://tempest/watersheds/trinity/trinity-transpo-area-v01/72h/docs/19790201.json):
> 
> ```json
> "metadata": {
>     ...
>     "watershed_name": "",
>     ...
>   }
> ```

The overwrite thing is done, but may not be necessary for Trinity as @ahdietrich is rerunning

**Comment by nick-j-roberts** [comment]
Format of ranked document was updated to minimize information being repeated from s3 docs and meilisearch upload function was revised to consume and construct meilisearch formatted documents from these ranked documents

**Comment by nick-j-roberts** [comment]
Closes #36

**Review by slawler — APPROVED** [review]

**Pull request merged**